slippi-stats.js

```diff
@@ -71,250 +76,311 @@ var final_player_name = user_player
+            end_more_kills = player_kills > opponent_kills
+            end_lower_percent = (player_kills == opponent_kills) && player_final_percent < opponent_final_percent
+
+            // disabled due to perceived inconsistencies
+            // try {
+            //     end_opponent_LRAS = (game.getGameEnd().lrasInitiatorIndex == opponent_num)
+            //     end_player_LRAS = (game.getGameEnd().lrasInitiatorIndex == player_num)
+            // }
+            // catch {
+            //     end_opponent_LRAS = false
+            //     end_player_LRAS = false
+            // } 
+    
+            // Every death is considered the opponent's kill
+            // If the player didn't quit out AND has more kills than the opponent, the same but with a lower percent, or the opponent quits out: it's a win, otherwise it's a loss. Ties handled above
+            //if (!end_player_LRAS && (end_more_kills || end_lower_percent || end_opponent_LRAS)) {
+            if (end_more_kills || end_lower_percent) {
+                console.log(`${index}: ${player_name || player_codes[player_num]} (${player_character}) beat ${opponent_name || opponent_code} (${opponent_character}) in ${game_length}!`)
+                data.total_wins = 1
+            } else {
+                console.log(`${index}: ${player_name || player_codes[player_num]} (${player_character}) lost to ${opponent_name || opponent_code} (${opponent_character}) in ${game_length}.`)
+            }
+            data.total_games = 1
+            data.player_character_num = player_character_num
+            data.player_name = player_name
+            data.opponent_code = opponent_code
+    
+            // Try to find last used nickname and actual connect code to display at the end
+            if (player_name.length > 0) {
+                final_player_name = player_name
+            }
+            data.real_player_code = player_codes[player_num]
+            if (opponent_arg && player_names[opponent_num]) {
+                if (opponent_name.length > 0) {
+                    final_opponent_name = opponent_name
+                }
+                real_opponent_code = player_codes[opponent_num]
+            }
+            data.game_seconds = game_seconds
+            return data
+        } catch (err) {
+            console.log(`${index}: Error reading replay (${file}). Ignoring results...`)
+            return data
         }
-        if (player_num == 'none') {
-            console.log(`${i}: User ${user_player} not found in replay. Ignoring results...`)
-            continue
-        }
-        if (opponent_arg && !opponent_found) {
-            console.log(`${i}: User ${opponent_player} not found in replay. Ignoring results...`)
-            continue
-        }
-        if (ignored_arg && ignored_opponent_found) {
-            console.log(`${i}: User ${found_ignored_opponent} found in replay. Ignoring results...`)
-            continue
+    }
+});
+
+(async () => {
```

Comment:
This is an anonymous async function so we can use `await` on the array of promises.